source/CCustomOpcodeSystem.cpp

```diff
@@ -2783,6 +2786,25 @@ namespace CLEO {
 		*thread << PLATFORM_WINDOWS;
 		return OR_CONTINUE;
 	}
+
+	//2000=2,%2s% = resolve_filepath %1s%
+	OpcodeResult __stdcall opcode_2000(CRunningScript* thread)
+	{
+		auto path = CLEO_ReadStringOpcodeParam(thread);
+		CLEO_ResolvePath(thread, path, MAX_STR_LEN);
+		CLEO_WriteStringOpcodeParam(thread, path);
+		return OR_CONTINUE;
+	}
+
+	//2001=2,%2s% = get_script_filename %1d%
+	OpcodeResult __stdcall opcode_2001(CRunningScript* thread)
+	{
+		CCustomScript* script;
+		*thread >> script;
+		if((int)script == -1) script = (CCustomScript*)thread; // current script
+		CLEO_WriteStringOpcodeParam(thread, script ? script->GetScriptFileName() : "(null)" );

```

Comment:
maybe instead of -1 being a current script, 0 will be it?
then

```
CCustomScript* script;
		*thread >> script;
		CLEO_WriteStringOpcodeParam(thread, script ? script->GetScriptFileName() : (CCustomScript*)thread->GetScriptFileName() );
```

Reply:
I thought about it, but then when somebody provide nullptr by mistake then function will pretend nothing happened.
I think -1 is used as special value in other opcodes too.

Reply:
Then we should make this opcode fallible (IF and SET) and return empty string and set condition to false if nullptr is given

Reply:
Hm, then we should go step forward, and search all scripts to match given pointer. This will prevent crash when random number is provided.

Reply:
this would be great, otherwise throw an error and halt the script

Reply:
or don't throw error and return false instead

Reply:
Updated